File: CS/E558/Form1.cs
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.UI;
using System.Windows.Forms;

namespace E558
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            schedulerControl1.EditAppointmentFormShowing+=schedulerControl1_EditAppointmentFormShowing;
        }

        private void schedulerControl1_EditAppointmentFormShowing(object sender, AppointmentFormEventArgs e)
        {
            MyAppointmentForm frm = new MyAppointmentForm(sender as SchedulerControl, e.Appointment, e.OpenRecurrenceForm);
            e.DialogResult = frm.ShowDialog();
            e.Handled = true;
        }
    }
    public class MyAppointmentForm : AppointmentForm
    {
        public MyAppointmentForm(SchedulerControl control, Appointment apt)
            : base(control, apt)
        {
        }
        public MyAppointmentForm(SchedulerControl control, Appointment apt,bool openRecurrenceForm)
            : base(control, apt, openRecurrenceForm)
        {
        }
        protected override Form CreateAppointmentRecurrenceForm(Appointment patternCopy, FirstDayOfWeek firstDayOfWeek)
        {
            XtraForm1 frm = new XtraForm1(patternCopy, firstDayOfWeek, this.Controller);
            frm.LookAndFeel.ParentLookAndFeel = this.LookAndFeel;
            frm.ShowExceptionsRemoveMsgBox = this.Controller.AreExceptionsPresent();
            return frm;
        }
    }
}

File: VB/E558/Form1.vb
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.UI
Imports System.Windows.Forms

Namespace E558
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
			AddHandler schedulerControl1.EditAppointmentFormShowing, AddressOf schedulerControl1_EditAppointmentFormShowing
		End Sub

		Private Sub schedulerControl1_EditAppointmentFormShowing(ByVal sender As Object, ByVal e As AppointmentFormEventArgs)
			Dim frm As New MyAppointmentForm(TryCast(sender, SchedulerControl), e.Appointment, e.OpenRecurrenceForm)
			e.DialogResult = frm.ShowDialog()
			e.Handled = True
		End Sub
	End Class
	Public Class MyAppointmentForm
		Inherits AppointmentForm

		Public Sub New(ByVal control As SchedulerControl, ByVal apt As Appointment)
			MyBase.New(control, apt)
		End Sub
		Public Sub New(ByVal control As SchedulerControl, ByVal apt As Appointment, ByVal openRecurrenceForm As Boolean)
			MyBase.New(control, apt, openRecurrenceForm)
		End Sub
		Protected Overrides Function CreateAppointmentRecurrenceForm(ByVal patternCopy As Appointment, ByVal firstDayOfWeek As FirstDayOfWeek) As Form
			Dim frm As New XtraForm1(patternCopy, firstDayOfWeek, Me.Controller)
			frm.LookAndFeel.ParentLookAndFeel = Me.LookAndFeel
			frm.ShowExceptionsRemoveMsgBox = Me.Controller.AreExceptionsPresent()
			Return frm
		End Function
	End Class
End Namespace